core/src/main/java/org/keycloak/representations/admin/v2/ClientRepresentation.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.representations.admin.v2;
+
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+
+import java.util.Set;
+
+public record ClientRepresentation(
+    @JsonPropertyDescription("ID uniquely identifying this client")
+    String clientId,
+
+    @JsonPropertyDescription("Human readable name of the client")
+    String displayName,
+
+    @JsonPropertyDescription("Human readable description of the client")
+    String description,
+
+    @JsonPropertyDescription("The protocol used to communicate with the client")
+    String protocol,
+
+    @JsonPropertyDescription("Whether this client is enabled")
+    Boolean enabled,
+
+    @JsonPropertyDescription("URL to the application's homepage that is represented by this client")
+    String appUrl,
+
+    @JsonPropertyDescription("URLs that the browser can redirect to after login")
+    Set<String> appRedirectUrls,
+
+    @JsonPropertyDescription("Login flows that are enabled for this client")
+    Set<String> loginFlows,
+
+    @JsonPropertyDescription("Authentication configuration for this client")
+    Auth auth,
+
+    @JsonPropertyDescription("Web origins that are allowed to make requests to this client")
+    String webOrigins,
+
+    @JsonPropertyDescription("Roles associated with this client")
+    Set<String> roles,
+
+    @JsonPropertyDescription("Service account configuration for this client")
+    ServiceAccount serviceAccount
+) {
+    public record Auth(
+            @JsonPropertyDescription("Whether authentication is enabled for this client")
+            Boolean enabled,
+
+            @JsonPropertyDescription("Which authentication method is used for this client")
+            String method,
+
+            @JsonPropertyDescription("Secret used to authenticate this client with Secret authentication")
+            String secret,
+
+            @JsonPropertyDescription("Public key used to authenticate this client with Signed JWT authentication")
+            String certificate
+    ) {}
+
+    public record ServiceAccount(
+            @JsonPropertyDescription("Whether the service account is enabled")
+            Boolean enabled,
+
+            @JsonPropertyDescription("Roles assigned to the service account")
+            Set<String> roles
+    ) {}
+}
```

Comment:
One major irk I have with the current REST API is the lack of information about which values are possibly `null`, this causes major headaches for the front-end. If possible all fields should be annotated with `@JsonProperty(required = true)`.
```suggestion
public record ClientRepresentation(
    @JsonProperty(required = true)
    @JsonPropertyDescription("ID uniquely identifying this client")
    String clientId,

    @JsonProperty(required = true)
    @JsonPropertyDescription("Human readable name of the client")
    String displayName,

    @JsonProperty(required = true)
    @JsonPropertyDescription("Human readable description of the client")
    String description,

    @JsonProperty(required = true)
    @JsonPropertyDescription("The protocol used to communicate with the client")
    String protocol,

    @JsonProperty(required = true)
    @JsonPropertyDescription("Whether this client is enabled")
    Boolean enabled,

    @JsonProperty(required = true)
    @JsonPropertyDescription("URL to the application's homepage that is represented by this client")
    String appUrl,

    @JsonProperty(required = true)
    @JsonPropertyDescription("URLs that the browser can redirect to after login")
    Set<String> appRedirectUrls,

    @JsonProperty(required = true)
    @JsonPropertyDescription("Login flows that are enabled for this client")
    Set<String> loginFlows,

    @JsonProperty(required = true)
    @JsonPropertyDescription("Authentication configuration for this client")
    Auth auth,

    @JsonProperty(required = true)
    @JsonPropertyDescription("Web origins that are allowed to make requests to this client")
    String webOrigins,

    @JsonProperty(required = true)
    @JsonPropertyDescription("Roles associated with this client")
    Set<String> roles,

    @JsonProperty(required = true)
    @JsonPropertyDescription("Service account configuration for this client")
    ServiceAccount serviceAccount
) {
    public record Auth(
            @JsonProperty(required = true)
            @JsonPropertyDescription("Whether authentication is enabled for this client")
            Boolean enabled,
 
            @JsonProperty(required = true)
            @JsonPropertyDescription("Which authentication method is used for this client")
            String method,

            @JsonProperty(required = true)
            @JsonPropertyDescription("Secret used to authenticate this client with Secret authentication")
            String secret,

            @JsonProperty(required = true)
            @JsonPropertyDescription("Public key used to authenticate this client with Signed JWT authentication")
            String certificate
    ) {}

    public record ServiceAccount(
            @JsonProperty(required = true)
            @JsonPropertyDescription("Whether the service account is enabled")
            Boolean enabled,

            @JsonProperty(required = true)
            @JsonPropertyDescription("Roles assigned to the service account")
            Set<String> roles
    ) {}
}
```